engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -282,7 +282,18 @@ public function __construct(Config $config) {
 		$this->setClassName('input', \Elgg\Http\Input::class);
 
 		$this->setFactory('imageService', function(ServiceProvider $c) {
-			$imagine = new \Imagine\Gd\Imagine();
+			$image_processor = _elgg_config()->image_processor;
+			switch ($image_processor) {
+				case 'imagick':
+					if (extension_loaded('imagick')) {
```

Comment:
And what if not?

Reply:
it'll fallback to GD

Reply:
I don't see how

Reply:
the break is only inside the if, otherwise it continues to the default case

Reply:
Ah, not very readable if you ask me

Reply:
that\s how PHP switches work ;)

Reply:
Adding breaks whereever?

Reply:
no the not having breaks and thus continuing to the next case

Reply:
But there is a break wrapped in an if statement

Reply:
FYI https://secure.php.net/manual/en/control-structures.break.php

Reply:
I know how switches work, I am saying this is not readable